File: PHP13-Sesiones/sesiones/saludo.php
<html>
<head>
<title>Saludo</title>
<meta charset="UTF-8" />
</head>
<body>
<?php

// Si no hay sesión iniciada, se mostrará un formulario para pedir nuestro nombre.
// Al enviar el formulario se iniciará una sesión almacenando el nombre como variable de sesión
// Si hay sesión iniciada, se mostrará el siguiente texto: Bienvenid@, Alberto*
$mensaje = "";

	if (session_status () == PHP_SESSION_NONE) {
			
	if ( isset ( $_POST ["enviar"] )) {
		
		session_start ();
		
		$_SESSION ['saludo'] = $_POST ['nombre'];
		
		$mensaje = "Bienvenid@ " . $_SESSION ['saludo'];
		
		
		if (isset ( $_REQUEST ["cerrarSesion"] )) {
			$_SESSION = array ();
			session_unset ();
			if (ini_get ( "session.use_cookies" )) {
				$params = session_get_cookie_params ();
				setcookie ( session_name (), '', time () - 42000, $params ["path"], $params ["domain"], $params ["secure"], $params ["httponly"] );
			}
			session_destroy ();
		}
	
	}else {
		
		?>
			<form action="<?php echo $_SERVER["PHP_SELF"]?>" method="post">
				<label>Introduzca su nombre:</label> <input type="text" name="nombre">
				<input type="submit" name="enviar" value="Enviar Nombre">
			</form>
			<?php
	}

}
?>
<h3><?php echo $mensaje;?></h3>
<p>
		<a href="<?php echo $_SERVER['PHP_SELF']."?cerrarSesion=true"?>">Cerrar
			sesión</a>
	</p>
</body>
</html>

File: Java13-Sesiones/src/servlets/Saludo.java
package servlets;

import java.io.IOException;
import java.io.PrintWriter;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

/**
 * Servlet implementation class Saludo
 */
@WebServlet("/Saludo")
public class Saludo extends HttpServlet {
	private static final long serialVersionUID = 1L;

	/**
	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse
	 *      response)
	 */
	protected void doGet(HttpServletRequest request, HttpServletResponse response)
			throws ServletException, IOException {
		HttpSession session = request.getSession();
		String mensajeError = "";
	
		if (request.getParameter("cerrarSesion")!=null) {
			session.invalidate();
			session=request.getSession();
		}
		
		if (request.getParameter("enviar") != null) {
			if (request.getParameter("usuario").equals("")) {
				mensajeError = "Usuario sin rellenar, complete el formulario antes de enviar.";
			} else
				session.setAttribute("usuario", request.getParameter("usuario"));
		}
		PrintWriter out = response.getWriter();
		out.println("<html><head><meta charset='UTF-8'/>" + "<style> .error {color: red}</style>"
				+ "<title>Sesiones en JavaEE</title></head><body>");
		response.setContentType("text/html;UTF-8");
		if (session.getAttribute("usuario")!=null) {
			out.println("<h3>Bienvenid@ "+session.getAttribute("usuario") +"</h3>");
		}
		else{
			out.print("<h3>Identificación del usuario</h3>");
			out.print("<form action='"+request.getRequestURI()+"' method='post'>");
			out.print("	<label>Usuario: </label></br><input type='text' name='usuario'> <br>");
			out.print("	<input type='submit' name='enviar' value='Enviar'>");
			out.print("</form>");	
			if (!mensajeError.equals("")) {
				out.println("<p class='error'>"+mensajeError+"</p>");
			}
		}
		
		out.println("<p><a href='" + request.getRequestURI() + "?cerrarSesion=true'>Cerrar sesión</a></p>");
		
		out.println("</body></html>");

	}

	/**
	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse
	 *      response)
	 */
	protected void doPost(HttpServletRequest request, HttpServletResponse response)
			throws ServletException, IOException {
		doGet(request, response);
	}

}
